ui/app/adapters/pki/pki-key-engine.js

```diff
@@ -0,0 +1,28 @@
+import ApplicationAdapter from '../application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+
+export default class PkiKeyEngineAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  optionsForQuery(id) {
+    let data = {};
+    if (!id) {
+      data['list'] = true;
+    }
+    return { data };
+  }
+
+  urlForQuery(backend, id) {
+    let url = `${this.buildURL()}/${encodePath(backend)}/keys`;
+    if (id) {
+      url = url + '/' + encodePath(id);
+    }
+    return url;
+  }
+
+  async query(store, type, query) {
+    const { backend, id } = query;
+    let response = await this.ajax(this.urlForQuery(backend, id), 'GET', this.optionsForQuery(id));
```

Comment:
I think we can just return here and remove the `async...await`?

Reply:
My understanding is that it's wrapping the ajax request in a promise—a good thing. It's essentially turning a sync call into an async call. I think it's a newer pattern for us. It's used in the named-path.js adapter query method and in the keymgmt/key adapter query method. Maybe a discussion for the team? I searched for some documentation on this, but couldn't find anything specific to our situation.

Reply:
Ahh okay - I had just peeked at some older `query` methods and from what I could see we only make it `async` if we need to manipulate or filter the response in some way. And that the error/promise handling happened in the route, in the model hook here for example - but I could definitely be missing something!

Reply:
I'll bring this up in our next standup. As I'm curious because the pattern isn't totally clear. Thanks for bringing it up.

Reply:
Good idea!